runtime-modules/content/src/lib.rs

```diff
@@ -1734,21 +1749,20 @@ decl_module! {
             // == MUTATION_SAFE ==
             //
 
-            // unreseve balance from previous bid
-            if let Some(bid) = eng_auction.top_bid.clone() {
-                if let Ok(ref old_bidder_account_id) = T::MemberAuthenticator::controller_account_id(bid.bidder_id) {
-                    T::Currency::unreserve(
-                        old_bidder_account_id,
-                        bid.amount
-                    );
-                }
-            }
+            if let Some(bid) = eng_auction.top_bid.as_ref() {
+                let bidder_account_id =
+                    T::MemberAuthenticator::controller_account_id(bid.bidder_id)?;
+                Self::withdraw_bid_payment(&bidder_account_id, bid.amount)?;
+            };
 
             let (updated_nft, event) = match eng_auction.buy_now_price {
                 Some(buy_now_price) if bid_amount >= buy_now_price => {
-
-                    // unfallible: can_reserve already called
-                    T::Currency::reserve(&participant_account_id, buy_now_price)?;
+                    // Make a new bid considering the "buy-now-price".
+                    Self::make_bid_payment(
+                        &participant_account_id,
+                        buy_now_price,
+                        None
+                    )?;
```

Comment:
I'm wondering what's the best way to handle functions that return `DispatchResult`, but we expect them to never fail.
Should we just ignore the result in that case, ie.:
```rust
                    let _ = Self::make_bid_payment(
                        &participant_account_id,
                        buy_now_price,
                        None
                    );
```
Or maybe just add a comment that the call is expected not to fail?
What do you think?

Reply:
I would suggest:
change `make_bid_payment` so that it returns nothing.
inside `make_bid_payment` use `let _ = ..` for `deposit` and `withdraw` subroutines, so we have non-fallible functions after the mutation mark and one has not to worry about possible errors. Then add a `/// Usable balance amount must be sufficient` doc line above the `make_bid_payment`

Reply:
If we have several calls of mutating functions that could return errors I'm not aware of good ways to deal with it until Parity implements transactional extrinsics (there were plans). 

There is an option of rolling back changes before the error - "compensation" that is used in some systems but it will clutter the code.

The comment ("can't fail") could introduce even more confusion because it will be outdated sooner or later and in general can't be bound to the check that guarantees the non-failure.

My rule of thumb is "fail fast" in such cases. If we introduce `let _ = fn_with_dispatch_result()` we could damage more state than necessary in case of failed method.